packages/compass-indexes/src/modules/indexes.ts

```diff
@@ -211,6 +229,25 @@ const _convertToModels = (indexes: Document[]): IndexDefinition[] => {
   });
 };
 
+function _mergeInProgressIndexes(
+  indexes: IndexDefinition[],
+  inProgressIndexes: IndexDefinition[]
+) {
+  for (const inProgressIndex of inProgressIndexes) {
+    const index = indexes.find((index) => index.name === inProgressIndex.name);
+
+    if (index) {
+      index.extra = index.extra ?? {};
+      index.extra.status = inProgressIndex.extra.status;
+      index.extra.error = inProgressIndex.extra.error;
```

Comment:
How do you feel about avoiding the mutation of the passed in index objects here? I've found that when things are mutated in methods it can lead to bugs down the line, as the changes to the objects can easily be hidden and unexpected.

Reply:
the indexes are cloned in the caller, but I like the extra safety

Reply:
changed